syntax = "proto3";

package qoin;

import "mediapipe/framework/formats/landmark.proto";
import "mediapipe/framework/formats/detection.proto";

service HandTracking {
  rpc HandTrackingPullStream(HandTrackingPullRequest)
      returns (stream HandTrackingPullReply) {}
  rpc HandTrackingPushStream(HandTrackingPushRequest)
      returns (stream HandTrackingPushReply) {}
}

message HandTrackingPullRequest {}

message HandTrackingPullReply {
  repeated mediapipe.Detection detection = 1;
  mediapipe.NormalizedLandmarkList landmark_list = 2;
}

message HandTrackingPushRequest {
  repeated mediapipe.Detection detection = 1;
  mediapipe.NormalizedLandmarkList landmark_list = 2;
}

message HandTrackingPushReply {}
